pkg/cloud/services/iamauth/reconcile.go

```diff
@@ -21,23 +21,17 @@ import (
 	"fmt"
 
 	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/aws/aws-sdk-go/service/iam"
 	"github.com/pkg/errors"
 	"k8s.io/klog/v2"
 
 	ekscontrolplanev1 "sigs.k8s.io/cluster-api-provider-aws/v2/controlplane/eks/api/v1beta2"
-	iamv1 "sigs.k8s.io/cluster-api-provider-aws/v2/iam/api/v1beta1"
 )
 
 // ReconcileIAMAuthenticator is used to create the aws-iam-authenticator in a cluster.
-func (s *Service) ReconcileIAMAuthenticator(ctx context.Context) error {
+func (s *Service) ReconcileIAMAuthenticator(ctx context.Context, nodeRoles map[string]struct{}) error {
```

Comment:
What should happen when `nodeRoles` has no items? Is that even possible?

Reply:
Currently, I'm defaulting to the old nodes role https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4011/files#diff-5d25b4431cab86a51260638407de8df3b6ee28e63ddef8c2c74e0e0a99c31d61R471

this was done to not break existing behavior

Reply:
Got it, thanks!